### PR TITLE
LW75 Support

### PR DESCRIPTION
**Pull request by Laneware**
LW75 Support

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/lw75/keymaps/iso/keymap.c [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS),
+
+	[2] = LAYOUT_iso(
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS),};
+
```
Comment (waffle87) [inline keyboards/lw75/keymaps/iso/keymap.c:31]:
```suggestion
		KC_LCTL, KC_LGUI, KC_LALT, KC_SPC, KC_RALT, KC_RGUI, KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT)
};

```

File: keyboards/lw75/keymaps/iso_split_backspace/keymap.c [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+	[2] = LAYOUT_iso_splitbs(
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS),};
+
+
```
Comment (waffle87) [inline keyboards/lw75/keymaps/iso_split_backspace/keymap.c:32]:
```suggestion
		KC_LCTL, KC_LGUI, KC_LALT, KC_SPC, KC_RALT, KC_RGUI, KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT)
};


```

File: keyboards/lw75/lw75.c [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+bool encoder_update_kb(uint8_t index, bool clockwise) {
+    if (!encoder_update_user(index, clockwise)) return false;
+    if (index == 0) { /* First encoder */
+        if (clockwise) {
+            tap_code(KC_VOLU);
+        } else {
+            tap_code(KC_VOLD);
+        }
+    } 
+        return false;
```
Comment (waffle87) [inline keyboards/lw75/lw75.c:15]:
```suggestion
        return true; 
```

File: keyboards/lw75/readme.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# LW-75
+
+![LW67](https://i.imgur.com/mcM1TTi.jpg)
```
Comment (waffle87) [inline keyboards/lw75/readme.md:3]:
```suggestion
![LW67](https://i.imgur.com/mcM1TTih.jpg)
```

File: keyboards/lw75/readme.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+A 75% Keyboard with a LHS rotary encoder, made and sold by Laneware Peripherals.
+
+* Keyboard Maintainer: [Laneware Peripherals](https://github.com/laneware)
+* Hardware Supported: LW-75
+* Hardware Availability: [Laneware Peripherals](https://lanewareperipherals.com/), 
+
+Make example for this keyboard (after setting up your build environment):
+
+    make lw75:default
```
Comment (waffle87) [inline keyboards/lw75/readme.md:13]:
```suggestion
    make lw75:default
    
Flashing example for this keyboard:
    make lw75:default:flash
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/lw75/readme.md (hunk 4 above)
Comment (Laneware) [reply keyboards/lw75/readme.md:3]: Rejected due to incorrect original link. New change includes your suggested change.

File: keyboards/lw75/lw75.c [hunk 6]
```diff
@@ -0,0 +1,17 @@
+// Copyright 2023 Laneware Peripherals
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "lw75.h"
+
+bool encoder_update_kb(uint8_t index, bool clockwise) {
```
Comment (drashna) [inline keyboards/lw75/lw75.c:6]:
```suggestion
#ifdef ENCODER_ENABLE
bool encoder_update_kb(uint8_t index, bool clockwise) {
```

File: keyboards/lw75/lw75.c [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+    if (!encoder_update_user(index, clockwise)) return false;
+    if (index == 0) { /* First encoder */
+        if (clockwise) {
+            tap_code(KC_VOLU);
+        } else {
+            tap_code(KC_VOLD);
+        }
+    } 
+        return true; 
+}
```
Comment (drashna) [inline keyboards/lw75/lw75.c:16]:
```suggestion
}
#endif
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/lw75/config.h [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include "config_common.h"
+
+/* key matrix size */
+#define MATRIX_ROWS 6
+#define MATRIX_COLS 16
+
```
Comment (waffle87) [inline keyboards/lw75/config.h:11]:
```suggestion
```

File: keyboards/lw75/info.json [hunk 9]
```diff
@@ -0,0 +1,28 @@
+{
+    "keyboard_name": "LW-75", 
+    "manufacturer": "Laneware Peripherals",
+    "url": "", 
+    "maintainer": "qmk", 
+    "usb": {
+        "vid": "0x4C50",
+        "pid": "0x1111",
+        "device_version": "0.0.1"
+    },
```
Comment (waffle87) [inline keyboards/lw75/info.json:10]:
```suggestion
    },
    "processor": "atmega32u4",
    "bootloader": "atmel-dfu",
```

File: keyboards/lw75/rules.mk [hunk 10]
```diff
@@ -0,0 +1,19 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
```
Comment (waffle87) [inline keyboards/lw75/rules.mk:6]:
```suggestion
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Review by waffle87 — APPROVED** [review]

File: keyboards/lw75/config.h [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+#include "config_common.h"
+
+/* key matrix pins */
+#define MATRIX_ROW_PINS { E6, B7, D0, D1, D2, B1 }
+#define MATRIX_COL_PINS { D3, D5, D4, D6, D7, B4, B5, B6, C6, C7, F7, F6, F5, F4, B0, B2 }
+
+/*ENCODER*/
+#define ENCODERS_PAD_A { F0 }
+#define ENCODERS_PAD_B { F1 }
+#define ENCODER_RESOLUTION 4
```
Comment (drashna) [inline keyboards/lw75/config.h:15]: This is the default behavior, and doesn't need to be defined. 
```suggestion
```

File: keyboards/lw75/config.h [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+
+/* COL2ROW or ROW2COL */
+#define DIODE_DIRECTION COL2ROW
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCE 5
+
```
Comment (drashna) [inline keyboards/lw75/config.h:22]: Also, the default and doesn't need defining.

Comment (Laneware) [reply keyboards/lw75/config.h:21]: Thanks, I'd rather leave it in for those who want to play around with the response time.

**Review by drashna — APPROVED** [review]

**Pull request merged**